openedx/features/course_experience/views/course_goals.py

```diff
@@ -0,0 +1,102 @@
+"""
+Helper functions that handle the creating, updating and deleting of course goals.
+"""
+import json
+
+from django.http import HttpResponse
+from django.utils.translation import ugettext as _
+from openedx.core.djangolib.markup import Text, HTML
+from enum import Enum
+from eventtracking import tracker
+
+from opaque_keys.edx.keys import CourseKey
+from student.models import CourseGoal
+
+
+class CourseGoalType(Enum):
+    """
+    Types of goals that a user can select.
+    """
+    CERTIFY = 'certify'
+    COMPLETE = 'complete'
+    EXPLORE = 'explore'
+    UNSURE = 'unsure'
+
+
+def get_course_goal_options():
+    return [CourseGoalType.CERTIFY.value, CourseGoalType.COMPLETE.value,
+            CourseGoalType.EXPLORE.value, CourseGoalType.UNSURE.value]
+
+
+def set_course_goal(request, course_id, goal=None):
+    """
+    Given a user, a course key and a particular goal, save the goal.
+    If no goal explicitly given, checks if a goal has been passed as
+    a data variable in an ajax call.
+    """
+    # If no explicit goal was passed in, try to grab it from the ajax request
+    if not goal and request.is_ajax():
+        goal = request.POST.get('goal')
+
+    # First ensure that the goal is an acceptable choice
+    if goal not in get_course_goal_options():
+        raise KeyError()
+
+    # Remove the existing course goal for the user
+    course_key = CourseKey.from_string(course_id)
+    remove_course_goal(request.user, course_key)
+
+    # Create and save course goal
+    new_goal = CourseGoal(user=request.user, course_key=course_key, goal=goal)
+    new_goal.save()
+
+    # Log the event
+    tracker.emit(
+        'edx.course.goal.added',
```

Comment:
@marcotuts: 
1. Are you ok with this event and its metadata?  Do you want this to be a research event?
2. Do you need this to go to GA, keeping in mind that we won't be able to track which goal?